src/rust/rust-proto/src/graplinc/grapl/api/plugin_registry/v1beta1_client.rs

```diff
@@ -1,226 +1,196 @@
 }
 
-#[async_trait::async_trait]
-impl Connectable for PluginRegistryServiceClient {
-    type Config = PluginRegistryClientConfig;
+impl client_impl::WithClient<PluginRegistryServiceClient<tonic::transport::Channel>>
+    for PluginRegistryClient
+{
     const SERVICE_NAME: &'static str =
         "graplinc.grapl.api.plugin_registry.v1beta1.PluginRegistryService";
 
-    #[tracing::instrument(err)]
-    async fn connect_with_endpoint(endpoint: Endpoint) -> Result<Self, ConnectError> {
-        let executor = Executor::new(ExecutorConfig::new(Duration::from_secs(30)));
-        let proto_client = create_proto_client!(
-            executor,
-            PluginRegistryServiceClientProto<tonic::transport::Channel>,
-            endpoint,
-        );
-
-        Ok(PluginRegistryServiceClient {
-            proto_client,
-            executor,
-        })
+    fn with_client(client: Client<PluginRegistryServiceClient<tonic::transport::Channel>>) -> Self {
+        Self { client }
     }
 }
 
-impl PluginRegistryServiceClient {
-    /// create a new plugin.
-    /// NOTE: Most consumers will want `create_plugin`, not `create_plugin_raw`.
-    /// NOTE: streaming RPCs aren't hooked up to client-executor just yet.
-    #[instrument(skip(self, request), err)]
-    pub async fn create_plugin_raw<S>(
-        &mut self,
-        request: S,
-    ) -> Result<native::CreatePluginResponse, GrpcClientError>
-    where
-        S: Stream<Item = native::CreatePluginRequest> + Send + 'static,
-    {
-        let proto_response = self
-            .proto_client
-            .create_plugin(request.map(proto::CreatePluginRequest::from))
-            .await
-            .map_err(Status::from)?;
-        let native_response = native::CreatePluginResponse::try_from(proto_response.into_inner())?;
-        Ok(native_response)
-    }
-
+impl PluginRegistryClient {
     /// Create a new plugin
-    ///
     #[instrument(skip(self, metadata, plugin_artifact), err)]
     pub async fn create_plugin<S>(
         &mut self,
         metadata: native::PluginMetadata,
         plugin_artifact: S,
-    ) -> Result<native::CreatePluginResponse, GrpcClientError>
+    ) -> Result<native::CreatePluginResponse, ClientError>
     where
         S: Stream<Item = Bytes> + Send + 'static,
     {
-        // Send the metadata first followed by N chunks
-        let request = futures::stream::iter(std::iter::once(
+        let proto_stream = futures::stream::iter(std::iter::once(
             native::CreatePluginRequest::Metadata(metadata),
         ))
-        .chain(plugin_artifact.map(native::CreatePluginRequest::Chunk));
+        .chain(plugin_artifact.map(native::CreatePluginRequest::Chunk))
+        .map(proto::CreatePluginRequest::from);
 
-        self.create_plugin_raw(request).await
+        self.client
+            .execute_streaming(proto_stream, |mut client, request| async move {
+                client.create_plugin(request).await
+            })
+            .await
```

Comment:
very slick and really impressed you got even the streaming case generalized :+1: